lib/utils/index.ts

```diff
@@ -349,25 +350,47 @@ function buildRegexpVisitor(
                 if (!patternNode || patternNode.type === "SpreadElement") {
                     continue
                 }
-                const pattern = getStringIfConstant(context, patternNode)
-                const flagsString = flagsNode
-                    ? getStringIfConstant(context, flagsNode)
-                    : null
+                const patternResult = getStaticValue(context, patternNode)
+                if (!patternResult || patternResult.value == null) {
+                    continue
+                }
+                let pattern: string | null = null
+                let { flagsString, ownsFlags } =
+                    flagsNode && flagsNode.type !== "SpreadElement"
+                        ? {
+                              flagsString: getStringIfConstant(
+                                  context,
+                                  flagsNode,
+                              ),
+                              ownsFlags: isStringLiteral(flagsNode),
+                          }
+                        : { flagsString: null, ownsFlags: false }
+                if (patternResult.value instanceof RegExp) {
+                    pattern = patternResult.value.source
+                    if (!flagsNode) {
+                        // If no flag is given, the flag is also cloned.
+                        flagsString = patternResult.value.flags
+                        ownsFlags = true
```

Comment:
Should that really be `true`?

`true` will be correct if we own the pattern but does `getStaticValue` guarantee that?

---

E.g. how will `no-useless-flag` react to this? The `g` flag is useless for `split` but should this even be reported?

```js
var foo = /a/g;
"abc".split(RegExp(foo));
export default foo;
```

Reply:
I think `new RegExp (foo)` can be considered to own the flag. If the flag is wrong, user can easily fix it.

I think it should be report and the user should fix it.

```js
var foo = /a/g;
"abc".split(RegExp(foo, ''));
export default foo;
```

Or, if the clone is wrong, fix it as follow:

```js
var foo = /a/g;
"abc".split(foo);
export default foo;
```

Reply:
You're right. I had the wrong idea of what it means to "own flags". I interpreted it as "`RegExp(foo)` own the `g` flag of `/a/g`" and not "`RegExp(foo)` own *its copy* of the `g` flag of `/a/g`".